sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeBatchActionsAsyncClient.java

```diff
@@ -425,4 +459,20 @@ private AnalyzeBatchActionsOptions getNotNullAnalyzeBatchActionsOptions(AnalyzeB
     private String getNotNullModelVersion(String modelVersion) {
         return modelVersion == null ? "latest" : modelVersion;
     }
+
+    private String[] parseActionErrorTarget(String targetReference) {
+        if (CoreUtils.isNullOrEmpty(targetReference)) {
+            throw logger.logExceptionAsError(new RuntimeException(
+                "Expected an error with a target field referencing an action but did not get one"));
+        }
+        // action could be failed and the target reference is "#/tasks/keyPhraseExtractionTasks/0";
+        final Pattern pattern = Pattern.compile(REGEX_ACTION_ERROR_TARGET, Pattern.MULTILINE);
+        final Matcher matcher = pattern.matcher(targetReference);
+        String[] taskNameIdPair = new String[2];
```

Comment:
NIT: I wonder if it will be worth creating an actual pair or a small internal struct that will save the action type and the id of it.
It will help when reading the code to understand what is in `[0]` and in `[1]`

Reply:
I think It is not worth. String[] should be good enough.